packages/relay/src/lib/clients/cache/localLRUCache.ts

```diff
@@ -123,10 +125,19 @@ export class LocalLRUCache implements ICacheClient {
    * @param {number} ttl - Time to live for the cached value in milliseconds (optional).
    * @param {string} requestIdPrefix - A prefix to include in log messages (optional).
    */
-  public set(key: string, value: any, callingMethod: string, ttl?: number, requestIdPrefix?: string): void {
-    const resolvedTtl = ttl ?? this.options.ttl;
-    this.logger.trace(`${requestIdPrefix} caching ${key}:${JSON.stringify(value)} for ${resolvedTtl} ms`);
-    this.cache.set(key, value, { ttl: resolvedTtl });
+  public async set(
+    key: string,
+    value: any,
+    callingMethod: string,
+    ttl?: number,
+    requestIdPrefix?: string,
+  ): Promise<void> {
+    return new Promise((resolve) => {
+      const resolvedTtl = ttl ?? this.options.ttl;
+      this.logger.trace(`${requestIdPrefix} caching ${key}:${JSON.stringify(value)} for ${resolvedTtl} ms`);
+      this.cache.set(key, value, { ttl: resolvedTtl });
+      resolve();
+    });
```

Comment:
Simply adding `async` to the method signature already makes this method return `Promise<void>` instead of `void`. 

When you don't explicitly return a `Promise` in an async method, the value you return will automatically be wrapped in a `Promise`. 

Thus, those changes are unneeded and we can revert this (same for other methods in this class where we haven't added anything additional to the behaviour of the methods).